README.md

```diff
@@ -1 +1,189 @@
-A simple CSV reader based on processing chunks of input.
\ No newline at end of file
+# ChunkedCSV.jl
+
+The main goal of this package is to offer a parser for CSV files that can be used for streaming purposes. 
+This means that it can work with data that is too large to fit into memory or in situations where we only need to process parts of the input data at a time. 
+The parser reads the data in chunks, and it can parse each chunk in parallel while simultaneously prefetching the next one to increase throughput. 
+Additionally, the parsed data is passed to a user-defined function that can consume it in any way required.
+
+Moreover, the package can automatically handle [`Parsers.jl`](https://github.com/JuliaData/Parsers.jl)-compatible types without dynamic dispatch overhead and with a low compilation overhead. 
+This is done without specializing on input schema, and the memory usage is upper bound by the user, who can control the chunk size. 
+All internally allocated buffers are reused, so if the number of values per MiB of input is roughly constant, allocations should be minimal.
+
+## Overview
+
+The package builds on [`ChunkedBase.jl`](https://github.com/JuliaData/ChunkedBase.jl), which provides a framework for parser packages, enabling them to customize three main aspects of the parsing process: how the data is parsed into a result buffer, what is the layout of the result buffer, and how the result buffer is consumed.
```

Comment:
I might add one sentence of explanation about what parts of this are handled in ChunkedBase so that a reader knows where to look (here or there) for whatever they're looking for. 
For example, I think you could just add this as one last sentence to this paragraph:

> ChunkedBase handles splitting a chunk by newlines, and then dividing the lines between N worker threads, where they are parsed and consumed by this package.

Or something like that?